src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -98,18 +105,30 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     String user = userService.getCurrentUser().getEmail();
     String text = Jsoup.clean(request.getParameter("text"), Whitelist.none());
-    /*replace file urls with corresponding html tags(<img>, <video>, <audio>)*/
+    
+  /*replace file urls with corresponding html tags(<img>, <video>, <audio>)*/
     String textReplaced = tagURLs(text);
     
-    /*adding image tags for the image file uploaded to Blobstore*/
+  /*adding image tags and corresponding lables for the image file uploaded to Blobstore to the end of message*/
     String messageText = textReplaced;
-    //get image url ulpoaded to Blobstore
-    List<String> imageBlobUrls = getUploadedFileUrl(request, "image");
-    //add image tage for uploaded image url at the end of message text
-    if(imageBlobUrls!=null ) {
-      for(String url:imageBlobUrls)
+    List<BlobKey> imageBlobKeys = getBlobKeys(request, "image");
+    /*get image url ulpoaded to Blobstore*/
+    List<String> imageBlobUrls = getUploadedFileUrl(imageBlobKeys);
+    /*add image tage for uploaded image url at the end of message text*/
+    if(imageBlobUrls!=null && !imageBlobUrls.isEmpty()) 
+    {
+      for(int i = 0; i<imageBlobUrls.size(); i++)
       {
-        messageText += "<img src=\"" + url + "\" />";   
+        //add image tag for the image user uploaded to Blobstore
+        messageText += "<img src=\"" + imageBlobUrls.get(i) + "\" />"; 
+        
+        //Get labels of the image and add it after image tag
+        byte[] blobBytes = getBlobBytes(imageBlobKeys.get(i));
+        List<EntityAnnotation> imageLabels = getImageLabels(blobBytes);
+        for(EntityAnnotation label : imageLabels)
```

Comment:
I think if ImageLabels is null, it will throw a NullPointerException in the for each loop.

Reply:
Yes, you're right! Thanks for letting me know, I will fix it